pkg/clusterconditions/mock/mock_test.go

```diff
@@ -68,39 +80,58 @@ func TestMock(t *testing.T) {
 			if logged.Method != "Valid" {
 				t.Errorf("logged method %q but expected Valid", logged.Method)
 			}
-			if !reflect.DeepEqual(logged.Condition, condition) {
-				t.Errorf("logged condition %v but expected %v", logged.Condition, condition)
+			if !reflect.DeepEqual(logged.Condition, testCase.condition) {
+				t.Errorf("logged condition %v but expected %v", logged.Condition, testCase.condition)
 			}
 		})
 	}
 
-	for i, testCase := range []struct {
+	matchTestCases := []struct {
+		condition     *configv1.ClusterCondition
+		pushMatches   []mock.MatchResult
 		expectedMatch bool
 		expectedError *regexp.Regexp
 	}{
 		{
+			condition:     nil,
+			pushMatches:   []mock.MatchResult{{Match: true, Error: nil}},
 			expectedMatch: true,
-			expectedError: nil,
```

Comment:
Possible nitpick:

I would maybe leave these explicit `nil` declarations (this line and previous line [91](https://github.com/openshift/cluster-version-operator/blob/fa8004488fbdc02dc58f004db09c7597c832b2bc/pkg/clusterconditions/mock/mock_test.go#L91)). They make it explicit that the expected value for the error in the respective test is `nil` as in the original code (since we are checking their values). 

But I don't have a strong feeling against removing them. Don't feel blocked by this.

Reply:
I have a slight preference for omitting these and relying on golang zero value defaults, it's idiomatic in go.

Reply:
Roger that, feel free to close this thread :+1: 

Meanwhile, I'll go re-read idioms in Go :books: